subprojects/scala/src/main/java/org/gradle/api/tasks/scala/ScalaDoc.java

```diff
@@ -133,6 +135,28 @@ public void setTitle(@Nullable String title) {
         this.title = title;
     }
 
+    /**
+     * Returns the amount of memory allocated to this task.
+     *
+     * @since 6.4
+     */
+    @Internal
+    @Nullable
+    public String getMaxMemory() {
```

Comment:
This should be exposed as a `Property<String>`

Reply:
Oh interesting. I found that most of properties in `javadoc` and `scaladoc` are not using `Property` provider. More than happy to change that, just want to know which one is the preferred way for consistency

Reply:
Yes, and migrating the existing things is going to be interesting. But the future is clearly with `Property<?>`, so please go ahead with that added one, leaving the other as they stand.

Reply:
Hi @ljacomet 

Moved this to property -> https://github.com/gradle/gradle/pull/12718/commits/93c6ed481d928987a391c7639db1aadfa5c2af8c

Also marked it as 6.5 since 6.4-rc-1 is already out 😅